grammar.ebnf

```diff
@@ -36,11 +38,40 @@ Statement = "{", {Statement}, "}"
 */
 LValue = Identifier, [ ".", Identifier ];
 
+Expression = ExpressionTerm 
+           | ParenthesisExpression
+           | FunctionCall
+           | NotExpression
+           | ArrayLookupExpression
+           | BinaryExpression
```

Comment:
Ok, that change is good. But this does not consider precedence.

Reply:
Well, this merits further conversation. Should we opt for the following order of precedence?
```
# High Precedence
|    * \ %
|    + - 
|  <   =<    >=    ==    !=
|    ||    && 
# Low Precedence
```
Also, wouldn't precedence levels invalidate the following rule you proposed, or did you suggest it as a stepping stone?
```
BinaryExpression = ExpressionTerm, BinOp, ExpressionTerm;
```

Reply:
> Well, this merits further conversation. Should we opt for the following order of precedence?

Sorry if that was overwhelming. Yeah, it seems good. Just make `||` have higher precedence than `&&`.

> Also, wouldn't precedence levels invalidate the following rule you proposed, or did you suggest it as a stepping stone?

Yes, as a stepping stone because the thought is similar when precedence is added.

Reply:
> Sorry if that was overwhelming.

Oh, worry not, I just like avoiding needless commits and changes ;)

Reply:
Okok :) Please do me a favor and add license / copyright notice in this file, with this template:
```cpp
/*
* Copyright: Copyright <YOUR_NAME> 2019.
* License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
* Authors:   <YOUR_NAME>
*/
```
You may open a separate PR for that or leave it to the next grammar PR.

Reply:
Isn't the copyright <YOUR_NAME> tag supposed to be for the project and not for the person writing it?

Reply:
Not in the way the Boost license has been written. You can have multiple copyright claims in different files. That's actually one of the reasons I chose BSL.